cmake/README.md

```diff
@@ -1,60 +1,243 @@
-   this step and use protobuf at your own risk.
-
-2. Use cmake to generate MSVC project files. Running the following commands
-   in a command shell will generate project files for Visual Studio 2008 in
-   a sub-directory named "build".
-
-        $ cd path/to/protobuf/cmake
-        $ mkdir build
-        $ cd build
-        $ cmake -G "Visual Studio 9 2008" ..
-
-   If you don't have gmock, skip the build of tests by turning off the
-   BUILD_TESTING option:
-
-        $ cmake -G "Visual Studio 9 2008" -DBUILD_TESTING=OFF ..
-
-3. Open the generated protobuf.sln file in Microsoft Visual Studio.
-4. Choose "Debug" or "Release" configuration as desired.
-5. From the Build menu, choose "Build Solution".  Wait for compiling to finish.
-6. If you have built tests, run tests.exe and lite-test.exe from a command
-   shell and check that all tests pass. Make sure you have changed the working
-   directory to the output directory because tests.exe will try to find and run
-   test_plugin.exe in the working directory.
-7. Run extract_includes.bat to copy all the public headers into a separate
-   "include" directory. This batch script can be found along with the generated
-   protobuf.sln file in the same directory.
-8. Copy the contents of the include directory to wherever you want to put
-   headers.
-9. Copy protoc.exe wherever you put build tools (probably somewhere in your
-   PATH).
-10. Copy libprotobuf.lib, libprotobuf-lite.lib, and libprotoc.lib wherever you
-    put libraries.
-
-  To avoid conflicts between the MSVC debug and release runtime libraries, when
-  compiling a debug build of your application, you may need to link against a
-  debug build of libprotobuf.lib.  Similarly, release builds should link against
-  release libs.
+This directory contains *CMake* files that can be used to build protobuf
+with *MSVC* on *Windows*. You can build the project as a from *Command Prompt*
+and using an *Visual Studio* IDE.
+
+You need to have [CMake](http://www.cmake.org), [Visual Studio](https://www.visualstudio.com)
+and optionaly [Git](http://git-scm.com) installed on your computer before proceeding.
+
+Most of the instruction will be given to the *Сommand Prompt*, but the same
+actions can be performed using appropriate GUI tools.
+
+Setting Environment
+===================
+
+From *Start* menu button open appropriate *Command Prompt*.
+
+For example: *Start* > *Visual Studio 2013* > *Visual Studio Tools* >
+*VS2013 x64 Native Tools Command Prompt*. You should see the *Command Prompt*:
+
+    C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin\amd64>
+
+Change current drive letter, if needed:
+
+    C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\bin\amd64>X:
+    X:\>
+
+Where *X* is your working drive letter. And change directory to your working folder:
+
+    X:\>cd Path\to
+    X:\Path\to>
+
+Where *Path\to* is path to your real working folder.
+
+Make folder where builded protobuf will be installed:
```

Comment:
Create a folder where protobuf headers/libraries/binaries will be installed after built.